src/js/components/Video/Video.js

```diff
@@ -80,6 +80,24 @@ const Video = forwardRef(
     const containerRef = useRef();
     const scrubberRef = useRef();
     const videoRef = useForwardedRef(ref);
+    const [controls, setControls] = useState(
```

Comment:
I think we should combine this `useState` and the subsequent `useEffect` into a single `useMemo`. We aren't using `setControls` anywhere else and the effect only depends on the `controlsProp`.

Reply:
Just switched it over to useMemo! Went over all the behavioral functionalities again and everything checked out!